### ISSUE TITLE
hledger-web "edit" mode do not assert "format" option of "commodity" directive

### ISSUE THREAD

**Issue by Simaex** [simonmichael/hledger#1194]
Hi
I have installed the latest hledger 1.16.1 using dastapov/hledger image
When I attempt to edit journal using "manage" access level of hledger-web I've got behavior that seem to be a bug. If I make any edits the journal file is not updated with error message that point to "format" option of commodity directive. In order to be able to edit journal from web interface I have to change commodity declaration to alternative syntax. Please find the examples:
journal header that prevents any edits:
```
Y2020  ; set default year to 2020
D $1,000.0 ;commodity-less amounts should be treated as dollars
commodity 10,000.0 $
; commodity SYMBOL
;   format EXAMPLEAMOUNT
; display Ukrainian hryvnia with currency name on the left,
; thousands comma-separated,
; period as decimal point, and zero decimal places.
commodity UAH
  format UAH 10,000,000.
; display EUROs with currency name on the left,
; thousands comma-separated,
; period as decimal point, and one decimal place.
commodity EUR
  format EUR 10,000,000.0
```
journal header that allow edits:
```
Y2020  ; set default year to 2020
D $1,000.0 ;commodity-less amounts should be treated as dollars
commodity 10,000.0 EUR
commodity 10,000.0 $
commodity 10,000. UAH
; commodity SYMBOL
;   format EXAMPLEAMOUNT
; display Ukrainian hryvnia with currency name on the left,
; thousands comma-separated,
; period as decimal point, and zero decimal places.
;commodity UAH
;  format UAH 10,000,000.
; display EUROs with currency name on the left,
; thousands comma-separated,
; period as decimal point, and one decimal place.
;commodity EUR
;  format EUR 10,000,000.0
```
The workaround is quite simple so this is not a priority to look into.

**Comment by simonmichael** [comment]
I can reproduce, starting hledger-web with `hledger-web --capabiliities=add,view,manage`, clicking the spanner icon to edit the journal, and trying to save gives an error like
```
Failed to load journal: .../a.j:10:10: | 10 | format UAH 10,000,000. | ^ commodity directive symbol "UAH " and format directive symbol "UAH" should be the same 
```
And odd one. Thanks for the report.

**Comment by Simaex** [comment]
I found a possibly related issue. I edit (actually commented) a transaction using spanner icon **Edit** . When I try to write the edit and error was thrown. I do not edit the line that is throwing assertion error. To write down the edit I have to comment the include directive, then edit the 2020.journal directly using nano editor to include once more.
Error:
```
Failed to load journal: /data/2020.journal:1:29: | 1 | include /data/rates.journal | ^ No existing files match pattern: /data/rates.journal
```
Sample journal (header only):
```
include /data/rates.journal
Y2020  ; set default year to 2020
D $1,000.0 ;commodity-less amounts should be treated as dollars
commodity 10,000.0 EUR
commodity 10,000.0 $
commodity 10,000. UAH
```
Both 2020.journal and rates.journal are in the same /data folder

Sample of rates.journal
```
P 2019-12-31 EUR $1.1219906807454056
P 2019-12-31 UAH $0.04220368214465608
P 2019-12-31 $ UAH23.694615
P 2020-01-01 EUR $1.1218295694081564
P 2020-01-01 UAH $0.04197857975802119
P 2020-01-01 $ UAH23.821673
P 2020-01-02 EUR $1.1170875293933655
P 2020-01-02 UAH $0.042203138900658875
P 2020-01-02 $ UAH23.69492
```

**Comment by simonmichael** [comment]
@Simaex could you try to to reduce this to a minimal example, and post the full procedure to reproduce it, as a new issue unless you think it's the same as this one. Thanks!

**Comment by simonmichael** [comment]
Or hold off until I fix the first issue. Data posted from the web has DOS line endings (`\r\n`), and our parsers don't like these unfortunately. When reading from disk, they are normalised automatically. I can convert them in the post handler, but then the file gets saved with unix line endings, which wouldn't be desirable on windows..

**Comment by Simaex** [comment]
@simonmichael I'll hold off since I am sure they are related. Thanks for your constant efforts!

**Comment by simonmichael** [comment]
@Simaex I believe this is fixed in #1194. Thanks for the report, good catch.

Here's some current line ending behaviour:

- hledger add and import commands will append with (at least some)
  unix line endings, possibly causing the file to have mixed line
  endings

- hledger-web edit and upload forms will write the file with
  the current system's native line endings, ie changing all
  line endings if the file previously used foreign line endings.

**Comment by Simaex** [comment]
Thanks a lot for fast fix! I'll try and come back if any issues arouse.
